### ISSUE TITLE
Feature Request - Automation of the Inventor's "Boost Modulation" Feat.

### ISSUE THREAD

**Issue by Unoblueboy** [foundryvtt/pf2e#11084]
Hi, currently if you pick up the Boost Modulation inventor feat in foundry, I can't find an easy way to change the damage type of your offensive boost. I've tried to create some rule elements to improve this but it isn't perfect. Here are the rule elements I've created (and under what feat):

<details>
<summary>Offensive Boost</summary>

```json
[
	{
		"choices": [
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Chill",
				"value": "cold"
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Ignition",
				"value": "fire"
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Jolt",
				"value": "electricity"
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Momentum",
				"value": "bludgeoning"
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Saws",
				"value": "slashing"
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Spike",
				"value": "piercing"
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Vitriol",
				"value": "acid"
			}
		],
		"key": "ChoiceSet",
		"prompt": "PF2E.SpecificRule.Inventor.OffensiveBoost.Prompt",
		"selection": "electricity",
		"rollOption": "offensive-boost:first"
	},
	{
		"damageType": "{item|flags.pf2e.rulesSelections.damage}",
		"diceNumber": 1,
		"dieSize": "d6",
		"key": "DamageDice",
		"predicate": [
			{
				"or": [
					{
						"and": ["feature:weapon-innovation", "item:id:{actor|flags.pf2e.innovationId}"]
					},
					{
						"and": ["feature:armor-innovation", "unarmed"]
					}
				]
			}
		],
		"selector": "strike-damage"
	},
	{
		"key": "ActiveEffectLike",
		"mode": "override",
		"path": "flags.pf2e.inventor.offensiveBoost.damageType",
		"value": "{item|flags.pf2e.rulesSelections.damage}",
		"priority": 60
	},
	{
		"alwaysActive": true,
		"key": "RollOption",
		"domain": "all",
		"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Prompt",
		"option": "damage",
		"suboptions": [
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Chill",
				"value": "cold",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:cold",
							"offensive-boost:second:cold",
							"offensive-boost:third:cold"
						]
					}
				],
				"selected": false
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Ignition",
				"value": "fire",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:fire",
							"offensive-boost:second:fire",
							"offensive-boost:third:fire"
						]
					}
				],
				"selected": false
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Jolt",
				"value": "electricity",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:electricity",
							"offensive-boost:second:electricity",
							"offensive-boost:third:electricity"
						]
					}
				],
				"selected": false
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Momentum",
				"value": "bludgeoning",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:bludgeoning",
							"offensive-boost:second:bludgeoning",
							"offensive-boost:third:bludgeoning"
						]
					}
				],
				"selected": true
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Saws",
				"value": "slashing",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:slashing",
							"offensive-boost:second:slashing",
							"offensive-boost:third:slashing"
						]
					}
				],
				"selected": false
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Spike",
				"value": "piercing",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:piercing",
							"offensive-boost:second:piercing",
							"offensive-boost:third:piercing"
						]
					}
				],
				"selected": false
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Vitriol",
				"value": "acid",
				"predicate": [
					{
						"or": [
							"offensive-boost:first:acid",
							"offensive-boost:second:acid",
							"offensive-boost:third:acid"
						]
					}
				],
				"selected": false
			}
		],
		"toggleable": true,
		"value": true
	}
]

```
</details>

<details>
<summary>Boost Modulation</summary>

```json
[
	{
		"choices": [
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Chill",
				"value": "cold",
				"predicate": [
					{
						"not": "offensive-boost:first:cold"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Ignition",
				"value": "fire",
				"predicate": [
					{
						"not": "offensive-boost:first:fire"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Jolt",
				"value": "electricity",
				"predicate": [
					{
						"not": "offensive-boost:first:electricity"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Momentum",
				"value": "bludgeoning",
				"predicate": [
					{
						"not": "offensive-boost:first:bludgeoning"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Saws",
				"value": "slashing",
				"predicate": [
					{
						"not": "offensive-boost:first:slashing"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Spike",
				"value": "piercing",
				"predicate": [
					{
						"not": "offensive-boost:first:piercing"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Vitriol",
				"value": "acid",
				"predicate": [
					{
						"not": "offensive-boost:first:acid"
					}
				]
			}
		],
		"flag": "damage",
		"key": "ChoiceSet",
		"prompt": "PF2E.SpecificRule.Inventor.OffensiveBoost.Prompt",
		"rollOption": "offensive-boost:second"
	},
	{
		"choices": [
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Chill",
				"value": "cold",
				"predicate": [
					{
						"not": "offensive-boost:first:cold"
					},
					{
						"not": "offensive-boost:second:cold"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Ignition",
				"value": "fire",
				"predicate": [
					{
						"not": "offensive-boost:first:fire"
					},
					{
						"not": "offensive-boost:second:fire"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Jolt",
				"value": "electricity",
				"predicate": [
					{
						"not": "offensive-boost:first:electricity"
					},
					{
						"not": "offensive-boost:second:electricity"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Momentum",
				"value": "bludgeoning",
				"predicate": [
					{
						"not": "offensive-boost:first:bludgeoning"
					},
					{
						"not": "offensive-boost:second:bludgeoning"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Saws",
				"value": "slashing",
				"predicate": [
					{
						"not": "offensive-boost:first:slashing"
					},
					{
						"not": "offensive-boost:second:slashing"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Spike",
				"value": "piercing",
				"predicate": [
					{
						"not": "offensive-boost:first:piercing"
					},
					{
						"not": "offensive-boost:second:piercing"
					}
				]
			},
			{
				"label": "PF2E.SpecificRule.Inventor.OffensiveBoost.Vitriol",
				"value": "acid",
				"predicate": [
					{
						"not": "offensive-boost:first:acid"
					},
					{
						"not": "offensive-boost:second:acid"
					}
				]
			}
		],
		"flag": "damage",
		"key": "ChoiceSet",
		"prompt": "PF2E.SpecificRule.Inventor.OffensiveBoost.Prompt",
		"rollOption": "offensive-boost:third"
	}
]

```

</details>
If this feature request is looked into further, it might want to be merged with #10960 as variable core has an option where you can change the damage type of offensive boost when reconfiguring.

**Comment by Unoblueboy** [comment]
This can also work with Persistent Boost if you add the following rule element to it
<details>
<summary>Persistent Boost</summary>

```json
{
  "key": "Note",
  "predicate": [
    {
      "or": [
        {
          "and": [
            "feature:weapon-innovation",
            "item:id:{actor|flags.pf2e.innovationId}"
          ]
        },
        {
          "and": [
            "feature:armor-innovation",
            "unarmed"
          ]
        }
      ]
    }
  ],
  "selector": "strike-damage",
  "text": "When you or your construct innovation damage a foe with offensive boost, that foe also takes @Damage[1d8[persistent, {actor|flags.pf2e.inventor.offensiveBoost.damageType}]] of the same damage type as the offensive boost damage.",
  "outcome": [],
  "title": "{item|name}"
}
```
</details>

**Comment by pedrogrullada** [comment]
Resolved by #13274
